include/ts/apidefs.h.in

```diff
@@ -279,7 +287,14 @@ typedef enum {
   TS_HTTP_CACHE_LOOKUP_COMPLETE_HOOK,
   TS_HTTP_PRE_REMAP_HOOK,
   TS_HTTP_POST_REMAP_HOOK,
+  TS_HTTP_REQUEST_BUFFER_READ_COMPLETE_HOOK,
```

Comment:
I suggest we change these to:
```
#define X(CMN) TS_HTTP_##CMN##_HOOK = TS_EVENT_##CMN - TS_EVENT_READ_REQUEST_HDR
X(READ_REQUEST_HDR), // defines TS_HTTP_READ_REQUEST_HDR_HOOK
X(OS_DNS), // defines TS_HTTP_OS_DNS_HOOK
// ...
X(REQUEST_BUFFER_READ_COMPLETE), // defines TS_HTTP_REQUEST_BUFFER_READ_COMPLETE_HOOK
#undef X
```

Reply:
I think short names like "X", even for a temporary `#define`, decreases readability.

Reply:
Then you should have written:  Myself, Alan, think short names like "X", even for indefinite article temporary #define, decrease readability.

Reply:
-1 on X too.

Reply:
I see the use of 'X' as I propose here as equivalent to using 'int i' as the index of a for loop.

Reply:
I guess you could use REBASE instead of X.  Still fairly short and somewhat suggestive of the purpose.

Reply:
REBASE it is. Merry Christmas, Walt! :)

In doing this, I had to re-order the events and hooks so that the hooks could reference the events via the macro now.